File: src/main/java/com/procurement/mdm/repository/CountryRepository.java
package com.procurement.mdm.repository;

import com.procurement.mdm.model.entity.Country;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CountryRepository extends JpaRepository<Country, Long> {

    List<Country> findAllById(Long id);

    List<Country> findCountriesByName(String name);

    List<Country> findCountriesByCode(String code);

}

File: src/main/kotlin/com/procurement/mdm/repository/CountryRepository.kt
package com.procurement.mdm.repository

import com.procurement.mdm.model.entity.Country
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
interface CountryRepository : JpaRepository<Country, String> {

    @Transactional(readOnly = true)
    fun findByCountryKeyLanguageCode(languageCode: String): List<Country>

    @Transactional(readOnly = true)
    fun findByCountryKeyLanguageCodeAndCountryKeyCode(languageCode: String, code: String): Country?

    @Transactional(readOnly = true)
    fun findByCountryKeyLanguageCodeAndName(languageCode: String, name: String): Country?

}
